### PR TITLE
Setup vault

### PR DESCRIPTION
**Pull request by rkondratowicz**
Will be used to setup secrets for tests (notify api key) and actual services on prod.

### CODE REVIEW

File: infrastructure/main.tf [hunk 1]
```diff
@@ -49,3 +53,15 @@ module "service" {
     SERVICE_BUS_CONNECTION_STRING = "${module.servicebus-topic.primary_send_and_listen_connection_string}/subscriptions/${local.subscription_name}"
   }
 }
+
+module "key-vault" {
```
Comment (nitinprabhu) [inline infrastructure/main.tf]: we could name this something like `private-beta-app-vault` ?

Comment (rkondratowicz) [reply infrastructure/main.tf]: as long as there's only one vault used the current name is clear imho.
The names of other modules in this file are: `servicebus-namespace`,  `servicebus-topic` etc.

File: infrastructure/main.tf [hunk 2]
```diff
@@ -2,6 +2,10 @@ provider "azurerm" {}
 
 locals {
   subscription_name = "pbi"
+
+  preview_vault_name     = "${var.raw_product}-aat"
+  default_vault_name     = "${var.raw_product}-${var.env}"
+  vault_name             = "${(var.env == "preview" || var.env == "spreview") ? local.preview_vault_name : local.default_vault_name}"
```
Comment (rkondratowicz) [inline infrastructure/main.tf]: following cmc convention: https://github.com/hmcts/cmc-claim-store/blob/master/infrastructure/main.tf#L16

File: infrastructure/main.tf (hunk 1 above)
Comment (lgonczar) [reply infrastructure/main.tf]: It all lives in the context of private beta app.

**Pull request merged**